packages/plugin-infra/src/hooks/useAvailablePlugins.ts

```diff
@@ -0,0 +1,17 @@
+import { useChainId, usePluginIDContext } from '../web3'
+import type { NetworkPluginID } from '../web3-types'
+import type { Plugin } from '../types'
+
+type HasRequirement = { enableRequirement: Plugin.Shared.Definition['enableRequirement'] }
+
+function checkPluginAvailable<T extends HasRequirement>(plugin: T, pluginId: NetworkPluginID, chainId: number) {
+    const supportedChainIds = plugin.enableRequirement.web3?.[pluginId]?.supportedChainIds
+    if (!supportedChainIds) return true
+    return supportedChainIds.includes(chainId)
+}
+
+export function useAvailabilePlugins<T extends HasRequirement>(plugins: T[]) {
```

Comment:
you don't need generics here. `plugins: HasRequirement[]`

Reply:
Need to make sure the return type is not just `HasRequirement[]`.